t/12_sles4sap_publicccloud.t

```diff
@@ -484,4 +485,57 @@ subtest '[get_hana_site_names] values from settings' => sub {
     ok(($res[1] eq 'ZhengHe'), "Value for the secondary site is from setting");
 };
 
+subtest '[wait_for_zypper] zypper unlocked at first try' => sub {
+    my $self = sles4sap_publiccloud->new();
+    my $pc_instance = Test::MockModule->new('publiccloud::instance');
+    my $instance = publiccloud::instance->new();
+    $pc_instance->redefine(run_ssh_command => sub { return 0; });
+
+    lives_ok { $self->wait_for_zypper($instance) } 'Zypper was not locked, command succeeded without retries';
+};
+
+subtest '[wait_for_zypper] zypper fails at first try with non 7 rc' => sub {
```

Comment:
What is expected behavior within `wait_for_zypper` and in the test in general if `zypper ref` return 1 ?

Reply:
Expected behavior is for `wait_for_zypper` to end without fail, as this function is specifically designed to wait in case zypper is locked (rc 7). If it fails for other reasons, this function lets the code proceed and fail in the test code.

Reply:
I do not see the `lets the code proceed`.

The function is not returning anything to the caller. So if the error is not 7 it stop looping and do nothing special.

So from the external the caller has no way to know if the function return for exit code 0 or exit code 1.

Am I correctly understand?

I think that (if it is like this) it is fine, and formally the function description does not deviate from this (so it is ok too)

```
Returns normally if Zypper is not locked, or dies after a maximum number of retries if Zypper remains locked.
```